cpp/Platform.Delegates/Platform.Delegates.MulticastDelegate.h

```diff
@@ -16,56 +15,29 @@ namespace Platform::Delegates
     template <typename ReturnType, typename... Args>
     class MulticastDelegate<ReturnType(Args...)> : public Delegate<ReturnType(Args...)>
     {
+    public:
         using DelegateRawFunctionType = ReturnType(Args...);
         using DelegateType = Delegate<DelegateRawFunctionType>;
         using DelegateFunctionType = std::function<DelegateRawFunctionType>;
 
-        mutable std::mutex mutex;
-        std::vector<DelegateType> callbacks;
-
-        void MoveCallbacksUnsync(MulticastDelegate &other)
-        {
-            callbacks = std::move(other.callbacks);
-        }
-
-        void CopyCallbacksUnsync(const MulticastDelegate &other)
-        {
-            callbacks = other.callbacks;
-        }
+        constexpr MulticastDelegate() noexcept = default;
 
-        void CopyCallbacks(const MulticastDelegate &other)
+        MulticastDelegate(const MulticastDelegate &multicastDelegate)
         {
-            // To prevent the deadlock the order of locking should be the same for all threads.
-            // We can use the addresses of MulticastDelegates to sort the mutexes. 
-            if (this < &other)
-            {
-                std::lock_guard lock1(mutex);
-                std::lock_guard lock2(other.mutex);
-                CopyCallbacksUnsync(other);
-            }
-            else
-            {
-                std::lock_guard lock1(other.mutex);
-                std::lock_guard lock2(mutex);
-                CopyCallbacksUnsync(other);
-            }
+            CopyCallbacks(multicastDelegate);
         }
 
-        void ClearCallbacks()
+        MulticastDelegate(MulticastDelegate &&multicastDelegate)
```

Comment:
You removed the const here. Why? Does it help with std::move?

Reply:
In c++ there is mainly two ways to pass parameters by reference:
```c++
MyClass(const MyClass& object);  // for copy semantic
MyClass(MyClass&& object);  // for move semantic
```
Other methods are considered bad practice. For example
```c++
MyClass(const MyClass&& object)
```
means, that it is temporary object, but you won't change it or get resources from it. For that case you can use simple 
```c++
const MyClass& object
```
form, and it can be used for both temporary objects and ordinary. Rvalue references should only be used if you want to grab resources from an object